e2e/select_test.go

```diff
@@ -486,6 +523,71 @@ func TestSelectCommand(t *testing.T) {
 	}
 }
 
+func TestSelectCommandEmptyBucket(t *testing.T) {
+	t.Parallel()
+
+	const (
+		region      = "us-east-1"
+		accessKeyID = "minioadmin"
+		secretKey   = "minioadmin"
+
+		query = "SELECT * FROM s3object s"
+	)
+
+	endpoint := os.Getenv(s5cmdTestEndpointEnv)
+	if endpoint == "" {
+		t.Skipf("skipping the test because %v environment variable is empty", s5cmdTestEndpointEnv)
+	}
+
+	type testcase struct {
+		name        string
+		cmd         []string
+		expectError bool
+	}
+
+	testcases := []testcase{
+		{
+			name: "input:json-lines,output:json-lines,all-versions:true,empty-bucket:true",
+			cmd: []string{
+				"select", "json",
+				"--all-versions",
+				"--query", query,
+			},
+			expectError: false,
+		},
+		{
+			name: "input:csv,output:csv,delimiter:comma,all-versions:true,empty-bucket:true",
+			cmd: []string{
+				"select", "csv",
+				"--all-versions",
+				"--query", query,
+			},
+			expectError: false,
+		},
+	}
+
+	for _, tc := range testcases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			bucket := s3BucketFromTestName(t)
+
+			s3client, s5cmd := setup(t, withEndpointURL(endpoint), withRegion(region), withAccessKeyID(accessKeyID), withSecretKey(secretKey))
+			createBucket(t, s3client, bucket)
+
+			src := fmt.Sprintf("s3://%s/", bucket)
+			tc.cmd = append(tc.cmd, src)
+			cmd := s5cmd(tc.cmd...)
+
+			result := icmd.RunCmd(cmd, withEnv("AWS_ACCESS_KEY_ID", accessKeyID), withEnv("AWS_SECRET_ACCESS_KEY", secretKey))
+
+			result.Assert(t, icmd.Success)
+			assert.DeepEqual(t, "", result.Stdout())
```

Comment:
```suggestion
                  if !tc.expectError {
                  	result.Assert(t, icmd.Success)
			assert.DeepEqual(t, "", result.Stdout())
                  }
```
Body of the test should utilize the `expectError`, please use it for correct testcase execution. Considering that `expectError` can be `true`, body should behave accordingly.  If you think that it's not required, `expectError` can be removed completely.